packages/counterfactual_nft/contracts/CounterfactualNftExt.sol

```diff
@@ -63,6 +63,20 @@ contract CounterfactualNftExt is WithCreator, CounterfactualNFT
         emit MintFromL2(to, id, amount, minter);
     }
 
+    function setContractURI(string memory contractURI_)
+        external
+        override
+    {
+        if (bytes(contractURI_).length > 0) {
+            if (_owner == address(0)) {             // every on can set when owner not initialized
```

Comment:
Can you explain a bit more how this would be used? Seems a bit strange to allow anyone to set this value.

Reply:
Allow NFTFactory to set this value before the owner is set. After owner is set, only owner can set this value.